.github/workflows/ci.yml

```diff
@@ -211,3 +212,21 @@ jobs:
         with:
           name: output
           path: _output/**
+
+      - name: Publish Artifacts to S3
+        run: make -j2 publish BRANCH_NAME=${GITHUB_REF##*/}
+        if: env.AWS_ACCESS_KEY_ID != ''
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: us-east-1
+
+      - name: Promote Artifacts in S3
+        if: github.ref == 'refs/heads/main' && env.AWS_ACCESS_KEY_ID != ''
+        run: make -j2 promote
+        env:
+          BRANCH_NAME: main
+          CHANNEL: main
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
```

Comment:
We have already injected the AWS_ACCESS_KEY_ID in the workflow environment variables (line 21). Do we still need to do again it here?

Reply:
Done.